docs/monitoring.md

```diff
@@ -0,0 +1,191 @@
+# Monitoring
+
+<!-- vim-markdown-toc GFM -->
+
+* [Installation](#installation)
+* [Monitoring in a separate namespace](#monitoring-in-a-separate-namespace)
+* [Using Custom Storage Classes](#using-custom-storage-classes)
+* [Troubleshooting](#troubleshooting)
+* [Monitoring without persistent disk](#monitoring-without-persistent-disk)
+* [Using custom storage classes](#using-custom-storage-classes-1)
+* [Accessing grafana](#accessing-grafana)
+* [Accessing prometheus](#accessing-prometheus)
+
+<!-- vim-markdown-toc -->
+
+
+## Installation
+
+The following instructions detail the installation of a monitoring system consisting
+of a Prometheus instance and corresponding Alert Manager in addition to a Grafana
+instance for viewing dashboards related to cluster and wire-services health.
+
+The monitoring system requires disk space if you wish to be resilient to pod
+failure. If you are deployed on AWS you may install the `aws-storage` helm
+chart which provides configurations of Storage Classes for AWS's elastic block
+storage (EBS). If you're not using AWS, instead of using `aws-storage`, you
+need to provide your [custom storage class](#using-custom-storage-classes).
+
+First we install the Storage Classes via the `aws-storage` chart:
+
+```
+helm upgrade --install --namespace demo demo-aws-storage charts/aws-storage \
+    -f values/wire-server/demo-values.yaml \
+    -f values/wire-server/demo-secrets.yaml \
+    --wait
+```
+
+Next we can install the monitoring suite itself
+
+There are a few known issues surrounding the `prometheus-operator` helm chart.
+
+You will likely have to install the Custom Resource Definitions manually before
+installing the `wire-server-metrics` chart:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/d34d70de61fe8e23bb21f6948993c510496a0b31/example/prometheus-operator-crd/alertmanager.crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/d34d70de61fe8e23bb21f6948993c510496a0b31/example/prometheus-operator-crd/prometheus.crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/d34d70de61fe8e23bb21f6948993c510496a0b31/example/prometheus-operator-crd/prometheusrule.crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/coreos/prometheus-operator/d34d70de61fe8e23bb21f6948993c510496a0b31/example/prometheus-operator-crd/servicemonitor.crd.yaml
+```
+
+Now we can install the metrics chart:
+
+```
+helm upgrade --install --namespace demo demo-wire-server-metrics charts/wire-server-metrics \
+    -f values/wire-server/demo-values.yaml \
```

Comment:
```suggestion
    -f values/wire-server-metrics/demo-values.yaml \
```

(and create that directory/file with some comments - pointing back to this repo would be fine, too)